python/pyspark/errors/error_classes.py

```diff
@@ -224,6 +224,11 @@
       "Argument `<arg_name>` should be a Column, int or str, got <arg_type>."
     ]
   },
+  "NOT_COLUMN_OR_LIST_OR_STR" : {
```

Comment:
I see there are a lot of `NOT_COLUMN_XXX` error classes, I am wondering is it possible to combine them into single one?

Reply:
Yes, we can either merge them into single error class as you mentioned, or create a representative main error class and add related type errors as sub-error classes under main error class what we do in JVM side.

I already created SPARK-43020 for address this issue, so let me do the refactoring once the basic migration is complete.